### ISSUE TITLE
[BUG] Windows not detecting GPU  in 0.6 causing models to crash on training

### ISSUE THREAD

**Issue by pingusix** [autogluon/autogluon#2446]
- [x] I have checked that this bug exists on the latest stable version of AutoGluon
- [ ] and/or I have checked that this bug exists on the latest mainline of AutoGluon via source installation

**Describe the bug**

Clean windows install as per instructions or upgrade from working 0.52 conda environment to 0.6  detects that GPU is available to Torch but running a simple test for a tabular training causes an error where it appears that the GPU is not detected causing training to fail.

**Expected behavior**

Training completes. 

**To Reproduce**
```python3
print('Torch GPU available: ',torch.cuda.is_available())  # Should be True
print(torch.cuda.device_count())
```
Returns:
```
Torch GPU available:  True
1
```

But:
```python3
import numpy as np
import pandas as pd
from autogluon.tabular import TabularDataset, TabularPredictor
import torch
train_data = TabularDataset('https://autogluon.s3.amazonaws.com/datasets/Inc/train.csv')
test_data = TabularDataset('https://autogluon.s3.amazonaws.com/datasets/Inc/test.csv')
#predictor = TabularPredictor(label='class',verbosity=4).fit(train_data=train_data)
predictor = TabularPredictor(label='class').fit(train_data=train_data,ag_args_fit={'num_gpus': 1})
```
Returns:
```
 Specified num_gpus per <Any Model> is more than the total: 0
```
It works if set num_gpus=0 or run using previous 0.52 conda environment 

**Screenshots**

**Installed Versions**
Which version of AutoGluon are you are using?  
If you are using 0.4.0 and newer, please run the following code snippet:
<details>

NSTALLED VERSIONS
------------------
date                   : 2022-11-19
time                   : 11:10:10.290181
python                 : 3.9.15.final.0
OS                     : Windows
OS-release             : 10
Version                : 10.0.22000
machine                : AMD64
processor              : Intel64 Family 6 Model 151 Stepping 2, GenuineIntel
num_cores              : 20
cpu_ram_mb             : 65277
cuda version           : None
num_gpus               : 1
gpu_ram_mb             : [8287]
avail_disk_size_mb     : None

accelerate             : 0.13.2
albumentations         : 1.1.0
autogluon.common       : 0.6.0
autogluon.core         : 0.6.0
autogluon.features     : 0.6.0
autogluon.multimodal   : 0.6.0
autogluon.tabular      : 0.6.0
autogluon.text         : 0.6.0
autogluon.timeseries   : 0.6.0
autogluon.vision       : 0.6.0
boto3                  : 1.26.13
catboost               : 1.1.1
dask                   : 2021.11.2
defusedxml             : 0.7.1
distributed            : 2021.11.2
evaluate               : 0.2.2
fairscale              : 0.4.6
fastai                 : 2.7.10
gluoncv                : 0.11.0
gluonts                : 0.11.1
hyperopt               : 0.2.7
joblib                 : 1.2.0
jsonschema             : 4.8.0
lightgbm               : 3.3.3
matplotlib             : 3.6.2
networkx               : 2.8.8
nlpaug                 : 1.1.10
nltk                   : 3.7
nptyping               : 1.4.4
numpy                  : 1.22.4
omegaconf              : 2.1.2
openmim                : None
pandas                 : 1.5.1
PIL                    : 9.0.1
psutil                 : 5.8.0
pytorch-metric-learning: None
pytorch_lightning      : 1.7.7
ray                    : 2.0.1
requests               : 2.28.1
scipy                  : 1.8.1
sentencepiece          : None
seqeval                : None
setuptools             : 65.5.0
skimage                : 0.19.3
sklearn                : 1.1.3
smart_open             : 5.2.1
statsmodels            : 0.13.5
text-unidecode         : None
timm                   : 0.5.4
torch                  : 1.12.0+cu113
torchmetrics           : 0.8.2
torchtext              : 0.13.0
torchvision            : 0.13.0+cu113
tqdm                   : 4.64.1
transformers           : 4.23.1
xgboost                : 1.7.1
Backend TkAgg is interactive backend. Turning interactive mode on.

</details>

**Additional context**

**Comment by vnghia** [comment]
I also had the same error when running `tabular` on Kaggle.

**Comment by yinweisu** [comment]
Hi, this is intended behavior starting ag 0.6 as we updated the logic to control resources.
You are expected to specify `num_gpus` when calling `fit()` to enable AG to use gpus.
You can refer to this [section](https://auto.gluon.ai/stable/tutorials/tabular_prediction/tabular-gpu.html#advanced-resource-allocation) of our doc to learn more about it

**Comment by Innixma** [comment]
Thanks for checking @pingusix , that does appear to be some kind of bug!

@yinweisu FYI. Also I thought if num_gpus is not specified, it would automatically use the GPUs available if relevant for the model. Why is it not detecting?

**Comment by Innixma** [comment]
@pingusix and @vnghia I believe this has been fixed in #2536 and is available in AutoGluon v0.6.1 release. Please give this a try and let us know if it works.

**Comment by pingusix** [comment]
Looks like the change has fixed it for me in 0.6.1 on Windows. Thanks.
